docs/en/platform-engineers/overview.md

```diff
@@ -1,3 +1,101 @@
+kind: Application
+metadata:
+  name: application-sample
+spec:
+  components:
+    - name: foo
+      type: worker # component type
+      settings:
+        image: "busybox"
+        cmd:
+        - sleep
+        - "1000"
+      traits:
+        - name: scaler
+          properties:
+            replicas: 10
+        - name: sidecar
+          properties:
+            name: "sidecar-test"
+            image: "nginx"
+    - name: bar
+      type: aliyun-oss # component type
+      bucket: "xxxxx"
+```
+
+Every `component` and `trait` in above abstraction is defined by platform team via `Definition` objects. For example, [`WorkloadDefinition`](https://github.com/oam-dev/kubevela/tree/master/config/samples/application#workload-definition) and [`TraitDefinition`](https://github.com/oam-dev/kubevela/tree/master/config/samples/application#scaler-trait-definition). As the end user, they only need to assemble these modules into an application. Also, if end user has any requirement, the platform team could customize module template in definitions by any time.
+
+#### A Unified Abstraction For All
+
+KubeVela intends to support any possible module type by natural, for example `CUE`, `Terraform`, `Helm`, etc and of course by a plain `Kubernetes CRD`. In order to define modules and parameters orgnized, we also introduced a [`catalog` structure](https://github.com/oam-dev/kubevela/blob/master/design/vela-core/APIServer-Catalog.md#catalog-structure). KubeVela will load such catalog via Git repo URL.
+
+```console
+/catalog/ # a catalog consists of multiple packages 
+|-- <package>
+    |-- v1.0 # a package consists of multiple versions
+        |-- metadata.yaml
+        |-- definitions/
+            |-- xxx-workload.yaml
+            |-- xxx-trait.yaml
+        |-- conditions/
+            |-- check-crd.yaml
+        |-- hooks/
+            |-- pre-install.yaml
+        |-- modules.yaml # could be helm, terraform, etc.
+    |-- v2.0
+|-- <package>
+```
+
+Hence, it's straightforward that you could use KubeVela to create unified abstraction that can deploy any kind of resource, including cloud services, as long as they could be encapsulated by a module and placed in the catalog above. Actually, in the `application-sample` above it defined a OSS bucket on Alibaba Cloud for the other component to consume, this is powered by Terraform module.
+
+#### No Configuration Drift
+
+Many of the existing modules today are defined by client side Infrastructure-as-Code (IaC) tools and even Kubernetes tool like Helm sits at client side as well. So in the nutshell, KubeVela encapsulation engine can just be implemented at client side which would be easier to be adopted.
+
+But client side abstractions, though light-weighted, always lead to a issue called infrastructure/configuration drift, i.e. the generated component instances are not in line with the expected configuration. This could be caused by incomplete coverage, less-than-perfect processes or emergency changes.
```

Comment:
```suggestion
But client side abstractions, though light-weighted, always lead to an issue called infrastructure/configuration drift, i.e. the generated component instances are not in line with the expected configuration. This could be caused by incomplete coverage, less-than-perfect processes or emergency changes.
```